File: API/Entities/Employee.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

public class Employee
{
        [ForeignKey("User")]

        public int EmployeeID { get; set; } // FK to Users
        public string? Position { get; set; }
        public decimal? Salary { get; set; }
        public bool IsAvailable { get; set; }
        public User User { get; set; }

}

File: client/src/app/_models/employee.ts
export interface User {
  userID?: number;
  name?: string;
  branchID?: number;
  branchName?: string;

}

export interface Employee {
  employeeID: number;
  position?: string;
  salary?: number;
  isAvailable?: boolean;
  // 🔹 Navigation to User
  user?: User;     // optional nested user object
}